maro/simulator/scenarios/vm_scheduling/cluster.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List
+
+from maro.backends.frame import NodeAttribute, NodeBase, node
+
+
+@node("clusters")
+class Cluster(NodeBase):
+    """Cluster node definition in frame."""
+    id = NodeAttribute("i2")
+    region_id = NodeAttribute("i2")
+    zone_id = NodeAttribute("i2")
+    data_center_id = NodeAttribute("i2")
+
+    # Total number of machines in the cluster.
+    total_machine_num = NodeAttribute("i")
+    # The number of empty machines in this cluster. A empty machine means that its allocated CPU cores are 0.
+    empty_machine_num = NodeAttribute("i")
+
+    def __init__(self):
+        self._id: int = 0
+        self._region_id: int = 0
+        self._zone_id: int = 0
+        self._data_center_id: int = 0
+
+        self._cluster_type: str = ""
+        self._rack_list: List[int] = []
+
+    def set_init_state(self, id: int, region_id: int, zone_id: int, data_center_id: int):
+        """Set initialize state, that will be used after frame reset.
+
+        Args:
+            id (int): Region id.
+        """
+        self._id = id
+        self._region_id = region_id
+        self._zone_id = zone_id
+        self._data_center_id = data_center_id
+
+        self.reset()
+
+    def reset(self):
+        """Reset to default value."""
+        self.id = self._id
+        self.region_id = self._region_id
+        self.zone_id = self._zone_id
```

Comment:
I am wondering whether we need to reset the ids in the reset (since it's only re-assign to the same value?), or just leave it  like the `cluster type`?

Reply:
The `reset` is used for assigning the value to the snapshot attributes from `set_init_state`.

Reply:
yes, but we only call `set_init_state()` once

Reply:
add comment for `reset()` function